docs/airframe-walkthrough.md

```diff
@@ -0,0 +1,1411 @@
+  .in(file("client"))
+  .enablePlugins(ScalaJSPlugin)
+  .settings(
+    name := "task-client",
+    libraryDependencies ++= Seq(
+      "org.wvlet.airframe" %%% "airframe-http"  % AIRFRAME_VERSION,
+      "org.wvlet.airframe" %%% "airframe-rx"    % AIRFRAME_VERSION,
+      "org.wvlet.airframe" %%% "airframe-log"   % AIRFRAME_VERSION
+    )
+  )
+  .dependsOn(api.js)
+
+// CLI module
+lazy val cli = project
+  .in(file("cli"))
+  .settings(
+    name := "task-cli",
+    libraryDependencies ++= Seq(
+      "org.wvlet.airframe" %% "airframe-launcher" % AIRFRAME_VERSION,
+      "org.wvlet.airframe" %% "airframe-http"     % AIRFRAME_VERSION,
+      "org.wvlet.airframe" %% "airframe-log"      % AIRFRAME_VERSION
+    )
+  )
+  .dependsOn(api.jvm)
+```
+
+**project/plugins.sbt**
+```scala
+addSbtPlugin("org.scala-js" % "sbt-scalajs" % "1.13.2")
+addSbtPlugin("org.portable-scala" % "sbt-scalajs-crossproject" % "1.3.2")
+addSbtPlugin("org.wvlet.airframe" % "sbt-airframe" % "(version)")
+```
+
+## Step 1: Foundation with Logging
+
+Let's start by setting up logging, which is essential for any application. Airframe provides [airframe-log](airframe-log.md), a modern logging library designed specifically for Scala.
+
+### Basic Logging Setup
+
+**api/src/main/scala/taskapp/api/Models.scala**
+```scala
+package taskapp.api
+
+import java.time.Instant
+
+case class Task(
+  id: String,
+  title: String, 
+  description: String,
+  completed: Boolean = false,
+  createdAt: Instant = Instant.now(),
+  updatedAt: Option[Instant] = None
+)
+
+case class CreateTaskRequest(title: String, description: String)
+case class UpdateTaskRequest(title: Option[String], description: Option[String], completed: Option[Boolean])
+case class TaskListResponse(tasks: Seq[Task], total: Int)
+```
+
+**server/src/main/scala/taskapp/server/TaskApp.scala**
+```scala
+package taskapp.server
+
+import wvlet.log.{LogSupport, Logger}
+
+object TaskApp extends App with LogSupport {
+  // Initialize the logger - this sets up console logging with source code locations
+  Logger.init
+  
+  info("Starting Task Management Application")
+  debug("Debug logging is available but won't show unless debug level is enabled")
+  
+  // Configure log level programmatically
+  // Logger.setDefaultLogLevel(LogLevel.DEBUG)
+  
+  info("Application initialized successfully")
+}
+```
+
+### Key Logging Features
+
+1. **Source Code Locations**: Automatically shows the file and line number where logs are generated
+2. **Macro-based Performance**: Debug logs have zero overhead when disabled
+3. **Programmatic Configuration**: No XML or property files needed
+
+Run the application:
+```bash
+$ sbt "server/run"
+```
+
+You'll see output like:
+```
+2023-12-15 10:30:15.123-0800  info [TaskApp] Starting Task Management Application  - (TaskApp.scala:8)
+2023-12-15 10:30:15.125-0800  info [TaskApp] Application initialized successfully  - (TaskApp.scala:14)
+```
+
+## Step 2: Configuration with airframe-config  
```

Comment:
We no longer recommend using airframe-config. Write examples without using .yml files and airframe-config module